plugins/woocommerce/client/admin/client/settings-payments/components/country-selector/test/country-selector.test.tsx

```diff
@@ -0,0 +1,92 @@
+/**
+ * External dependencies
+ */
+import '@testing-library/jest-dom/extend-expect';
+import { render, screen, fireEvent } from '@testing-library/react';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { CountrySelector } from '../country-selector';
+
+describe( 'CountrySelector', () => {
+	// Sample items for testing
+	const mockItems = [
+		{ key: 'US', name: 'United States' },
+		{ key: 'CA', name: 'Canada' },
+		{ key: 'FR', name: 'France' },
+	];
+
+	const mockOnChange = jest.fn();
+
+	const defaultProps = {
+		name: 'country-selector',
+		label: __( 'Select a country', 'woocommerce' ),
+		options: mockItems,
+		onChange: mockOnChange,
+		value: mockItems[ 0 ], // Initially selected item
+		placeholder: __( 'Choose a country', 'woocommerce' ),
+	};
+
+	afterEach( () => {
+		jest.clearAllMocks();
+	} );
+
+	it( 'renders correctly with initial props', () => {
+		render( <CountrySelector { ...defaultProps } /> );
+		expect( screen.getByText( 'Select a country' ) ).toBeInTheDocument();
+		expect( screen.getByText( 'United States' ) ).toBeInTheDocument();
+		expect( screen.getByRole( 'combobox' ) ).toBeInTheDocument();
+	} );
+
+	it( 'opens the dropdown menu when the button is clicked', () => {
+		render( <CountrySelector { ...defaultProps } /> );
+		const toggleButton = screen.getByRole( 'combobox' );
+
+		fireEvent.click( toggleButton );
+		expect( screen.getByRole( 'listbox' ) ).toBeInTheDocument();
+		expect( screen.getByText( 'Canada' ) ).toBeInTheDocument();
+		expect( screen.getByText( 'France' ) ).toBeInTheDocument();
```

Comment:
Is US not expected to be on the list?

Reply:
It wasn’t added intentionally, as the currently selected country is the United States, which is already displayed on the page.